### ISSUE TITLE
Should this work?

### ISSUE THREAD

**Issue by luandro** [SkyzohKey/react-native-apk#2]
Hey, this is an awesome package. Is it still maintained?

I've tried installing it, but it doesn't even get imported...

**Comment by SkyzohKey** [comment]
It does work and I'm using it in the M-Droid app, feel free to check the code. Also did you ran react-native link `react-native-apk` ?

**Comment by luandro** [comment]
Yea, jus re-did the whole process just to make sure. Error:

```
E ReactNativeJS: TypeError: undefined is not an object (evaluating '_reactNativeApk2.default.getNonSystemApps')
```

That's because you're not exporting those functions in your [index.js](https://github.com/SkyzohKey/react-native-apk/blob/master/index.js). I changed so that they did, but still the `NativeModules.ReactNativeAPK` isn't working. It's something in the Java side of things. Can you help me debug it?

**Comment by hanssonfredrik** [comment]
@luandro did you get this to work?

**Comment by hanssonfredrik** [comment]
@SkyzohKey I am getting the same error as @luandro. Any idea?
React native 0.54.4

**Comment by luandro** [comment]
Nope @hanssonfredrik. If I really come to need it I'll try copy/pasting the Java part directly to my project. I think some changes are needed for this to work with a `npm i`.

**Comment by SkyzohKey** [comment]
I'll fix the issue in the week guys, don't worry ;)

**Comment by hanssonfredrik** [comment]
Great news!

**Comment by luandro** [comment]
That would be awesome @SkyzohKey 👍

**Comment by hanssonfredrik** [comment]
@SkyzohKey do you have any status update here? Anything that I can help you with?

**Comment by SkyzohKey** [comment]
Sorry for the delay very busy at work. I will push something today :)

**Comment by SkyzohKey** [comment]
@luandro @hanssonfredrik Issue should be fixed as of now, can you confirm? Will push a new NPM package if this is solved. :)

**Comment by hanssonfredrik** [comment]
I get "undefined is not an object (evaluating 'RN.NativeModules')

There is some breaking changes in React Native 0.47 that forces you to remove the @override statements in the ReactNativeAPKPackage.java as well. I tried to remove these but still get the same error. 

I am running on React Native 0.55.4.

**Comment by hanssonfredrik** [comment]
I removed the .default on the row below, and then the "undefined" error disappeared.

from:
`var RN = require("react-native").default;
`
to:
`var RN = require("react-native");
`

But, when executing the code below: 

`ReactNativeAPK.installApp(path);`

I get the following error:

Possible Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating RNAPK.installApp).

**Comment by SkyzohKey** [comment]
Mhn that's really weird because on my side it works. Can you make a repoductible test repository so that I can further investigate please?

**Comment by hanssonfredrik** [comment]
@SkyzohKey what version of react native are you testing this with?

**Comment by hanssonfredrik** [comment]
I had of course forgot to link the package... 
I get the error "there was a problem parsing the package" when running the installPackage function. I read something that it could be the wrong name of the APK file, but I have verified that it is correct.

EDIT: Found the problem here as well. I am using react-native-fs to download the app locally. Problem was that I was saving the app to the document directory. When changing this to ExternalDirectoryPath everything worked as supposed. Thanks!
